experiment/stop_experiment.py

```diff
@@ -38,17 +38,18 @@ def stop_experiment(experiment_name, experiment_config_filename):
     gce.initialize()
     instances = list(gce.get_instances(cloud_project, cloud_compute_zone))
 
-    trial_prefix = 'r-' + experiment_name
-    experiment_instances = [
-        instance for instance in instances if instance.startswith(trial_prefix)
-    ]
+    experiment_instances = []
     dispatcher_instance = experiment_utils.get_dispatcher_instance_name(
         experiment_name)
     if dispatcher_instance not in instances:
         logger.warning('Dispatcher instance not running, skip.')
     else:
         experiment_instances.append(dispatcher_instance)
 
+    trial_prefix = 'r-' + experiment_name
```

Comment:
This is important since dispatcher needs to be killed first to avoid race conditions with instance deletes.

Reply:
ACK